src/database/schema/ChannelListSchema.ts

```diff
@@ -299,29 +390,55 @@ class ChannelList implements BaseDbSchema {
     });
   }
 
-  static fromAnonymousChannelAPI(data: AnonymousChannelData): ChannelList {
+  static fromChannelAPI(data: ChannelData, channelType: ChannelType): ChannelList {
+    const isPM = channelType === 'PM';
+    const firstMessage = data?.firstMessage;
+    const isSystemMessage = firstMessage?.type === 'system' || firstMessage?.isSystem;
+    const isMe = firstMessage?.user?.id === data?.myUserId;
+    let descriptionSystemMessage;
+
+    if (isPM && isMe && isSystemMessage) descriptionSystemMessage = firstMessage?.textOwnMessage;
+
     return new ChannelList({
       id: data?.id,
       channelPicture: data?.targetImage,
       name: data?.targetName,
-      description: data?.firstMessage?.text || data?.firstMessage?.message || '',
-      unreadCount: 0,
-      channelType: 'ANON_PM',
+      description: descriptionSystemMessage || firstMessage?.text || firstMessage?.message || '',
+      unreadCount: data?.unreadCount ?? 0,
+      channelType,
       lastUpdatedAt: data?.last_message_at,
-      lastUpdatedBy: data?.firstMessage?.user?.id,
+      lastUpdatedBy: firstMessage?.user?.id,
       createdAt: data?.created_at,
       rawJson: data,
       user: null,
       members: null
     });
   }
 
-  static fromAnonymousPostNotificationAPI(data: AnonymousPostNotification): ChannelList {
+  static fromSignedPostNotificationAPI(data: SignedPostNotification): ChannelList {
     return new ChannelList({
       id: data?.activity_id,
       channelPicture: data?.postMaker?.data?.profile_pic_url || '',
       name: data?.titlePost,
       description: data?.titlePost,
+      unreadCount: data?.unreadCount ?? 0,
+      channelType: 'POST_NOTIFICATION',
+      lastUpdatedAt: data?.data?.updated_at,
+      lastUpdatedBy: '',
+      createdAt: new Date().toISOString(),
+      rawJson: data,
+      user: null,
+      members: null,
+      expiredAt: data?.expired_at
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `fromSignedPostNotificationAPI` function seems to be implemented correctly. It creates a new `ChannelList` object from the provided data and sets the `channelType` to 'POST_NOTIFICATION'.



<!-- This is an auto-generated comment by OSS CodeRabbit -->